src/compiler/scala/tools/nsc/backend/jvm/BCodeBodyBuilder.scala

```diff
@@ -530,6 +530,7 @@ abstract class BCodeBodyBuilder extends BCodeSkelBuilder {
               if (cast) {
                 mnode.visitTypeInsn(asm.Opcodes.NEW, jlClassCastExceptionRef.internalName)
                 bc dup ObjectRef
+                mnode.visitMethodInsn(asm.Opcodes.INVOKESPECIAL, jlClassCastExceptionRef.internalName, INSTANCE_CONSTRUCTOR_NAME, "()V", true)
```

Comment:
FWIW, Scala.js [has corresponding code](https://github.com/scala-js/scala-js/blob/99e6017f55427d83fb1cd35c55def9ed8e875e11/compiler/src/main/scala/org/scalajs/core/compiler/GenJSCode.scala#L2344-L2352) in its codegen, but I'm not sure it is ever exercised ^^